### PR TITLE
feat/custom gas token: fix OptimismPortal contract size

### PR DESCRIPTION
**Pull request by 0xfuturistic**

### CODE REVIEW

File: packages/contracts-bedrock/src/L1/OptimismPortal.sol [hunk 1]
```diff
@@ -366,17 +366,13 @@ contract OptimismPortal is Initializable, ResourceMetering, ISemver {
...
-                uint256 startBalance = IERC20(token).balanceOf(address(this));
-
                 // Transfer the ERC20 balance to the target, accounting for non standard ERC20
                 // implementations that may not return a boolean. This reverts if the low level
                 // call is not successful.
                 IERC20(token).safeTransfer({ to: _tx.target, value: _tx.value });
 
                 // The balance must be transferred exactly.
-                if (IERC20(token).balanceOf(address(this)) != startBalance - _tx.value) {
+                if (IERC20(token).balanceOf(address(this)) != _balance) {
```
Comment (tynes) [inline packages/contracts-bedrock/src/L1/OptimismPortal.sol:375]: This is not safe, anybody can `ERC20.transfer` to this contract out of band and then brick deposits

Comment (0xfuturistic) [reply packages/contracts-bedrock/src/L1/OptimismPortal.sol:375]: totally. maybe we should use `if (IERC20(token).balanceOf(address(this)) < _balance) { revert(); }`

Comment (0xfuturistic) [reply packages/contracts-bedrock/src/L1/OptimismPortal.sol:375]: also, how is this different from the previous approach? looks like the contract could be bricked without any of the changes in this PR

Comment (tynes) [reply packages/contracts-bedrock/src/L1/OptimismPortal.sol:375]: If you can produce a proof of concept that bricks the existing implementation, that would be amazing :)

**Pull request closed without merging**